harvard-lts/mps-embed
# Switching over to nodejs Harvard version.

**Switching over to nodejs Harvard version.**
* * *

**JIRA Ticket**: [https://jira.huit.harvard.edu/browse/LTSVIEWER-93](LTSVIEWER-93)

# What does this Pull Request do?
This deletes all the Stanford code and replaces it with a very basic oembed API for use with Harvard Library items.
It uses NodeJS / Express instead of Ruby on Rails. 

There is a homepage with a few examples, but other URNs could be plugged into the api to get oembed responses.

# How should this be tested?

A description of what steps someone could take to:
* Completely rebuild your container: `docker-compose -f docker-compose-local.yml up -d --build --force-recreate`
* Go to: https://localhost:23018/ You should see a homepage similar to the MPS Viewer, except it is branded as MPS Embed.
* Clicking on any of the links on the homepage will take you to the API which returns a JSON oEmbed response. It should contain the correct title, iiif manifest url, viewer url and an iframe tag with the item.
* **NOTE:** Unfortunately right now the browser seems to cache the response, so you may need to hit refresh to see the correct response from the API.
* You can take the URL inside the <iframe> and paste it into your browser to confirm it is the right item. For example, this API call https://localhost:23018/api??recordIdentifier=W401827_URN-3:HUL.ARCH:2009747 has [this ](https://viewer-qa.lib.harvard.edu/viewer/ids/10274484\) viewer url in the `<iframe>` which shows the correct item.
* **NOTE:** Right now the viewer urls are hard-coding `viewer-qa`. I will iterate on this later.
* You should be able to plug in other URNs into the request variable of the API, but I haven't done very much testing on this yet....


# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? No 
- integration tests? No

# Interested parties
@enriquediaz @mferrarini @awoods
